### ISSUE TITLE
Define the boundary between country-specific code and generic downstream pipeline.

### ISSUE THREAD

**Issue by andrewphilipsmith** [Urban-Analytics-Technology-Platform/poppusher#38]
Initially, I suggest working with the following three requirements:

1. There should be three assets produced for each country, with standard names:
```
{ISO3}/census_geometry
{ISO3}/census_demographics
{ISO3}/metadata
```

2. There should be a single job that materializes all of the assets for a single country.
3. There should be a single asset group for the country, which is the ISO3 code.

Provided those three standard assets exist, the implementation is free to include as few or as many intermediate, country-specific assets as is necessary.

There are some shortcomings with this proposal (broadly similar to some of the [considerations I raised here](https://github.com/Urban-Analytics-Technology-Platform/popgetter/issues/34#issuecomment-1819801039)) - notably how to cope with multiple data sources per country. However, I suggest trying it as is (given we have a reasonably small number of countries at present) and seeing what we learn.

**Comment by andrewphilipsmith** [comment]
Quick notes/tasks:

- [ ] This was explored during last Friday's hackathon. More details about this need to be added (either in this issue, or as a separate linked PR).
- [ ] When completing this, we should review the use of native Python `docker` API vs [Dagster's docker wrapper](https://docs.dagster.io/_apidocs/libraries/dagster-docker). The reasons for our choice should be added somewhere appropriate in the contributing docs or comments.

**Comment by andrewphilipsmith** [comment]
Moved from #68:

- [ ] Upload and publish metadata
- [ ] Change the partition-key for the publishing pipeline, so that it abstracts the country-specific naming of tables/variables.
- [ ] Implement a general-purpose interface to describe the boundary between this and the country-specific pipelines

**Comment by andrewphilipsmith** [comment]
Closed by #84, #92, #102 and #113.

These all standardise the outputs of the country-level pipelines.
